### DISCUSSION TITLE
how do you handle dates or timestamp ?

### DISCUSSION THREAD

**Discussion by euri10** [DanCardin/cappa#122] [Q&A] [answered]
Currently I'm using the parse kwarg to do that, I just wondered if this was the "intended" way to use it
```
def parse_date(date_str: str) -> datetime.date:
    return datetime.datetime.strptime(date_str, "%Y-%m-%d").date()

@cappa.command()
@dataclass
class TotoArgs:
    start: Annotated[datetime.date| None, cappa.Arg(short=False, long=True, parse=parse_date)] = None
    end: Annotated[datetime.date | None, cappa.Arg(short=False, long=True, parse=parse_date)] = None

```

**Comment by DanCardin** [comment, marked as answer]
I personally have done `pendulum.parse` for all my stuff. Although I think it's probably worth adding a specific inference handler for exactly this/ and `datetime`, just because they're built in and obvious

EDIT: merged in handlers for date/datetime/time
